NetKVM/Common/ParaNdis_Common.cpp

```diff
@@ -143,6 +144,7 @@ static const tConfigurationEntries defaultConfiguration =
     { "*UsoIPv4", 1, 0, 1},
     { "*UsoIPv6", 1, 0, 1},
 #endif
+    {"MaxFragmentsInOneNB", MAX_FRAGMENTS_IN_ONE_NB, 16, 256},
```

Comment:
You do not modify the INF, so we do not have any easy way to change the setting in the registry. One can populate it to the registry manually but it is doubtful that eny will do it ever.
Let's understand together what is the use case for this configurability? Do we need it or not. If not, we can just assign pConfiguration->MaxFragmentsInOneNB = MAX_FRAGMENTS_IN_ONE_NB

Reply:
@zjmletang
If you really want some flexibility, it worth to make it configurable via INF, then you can use netkvmco.exe to change this parameters and apply the change via command line. You also can use the driver off the shelf (I mean with the default define of 256) and use netkvmco to set the limit via command line the same way.
But for me it looks kind of clumsy to have define of max number of fragment as 64 and allow it to be bigger at the same time.
If your hardware capability will be changed - how you can recognize what are exact capabilities of the adapter at run time. This indeed requires some change/addition in the spec, like some additional capability via PCI configuration.
In your downstream driver you can put everything you want (like automatically derive this exact capability via PCI revision). But in the upstream driver this is kind of hidden backdoor configuration.

Reply:
@ybendito ,Okay, I'll go ahead and remove it since it doesn't seem to need configuration. In fact, we generally wouldn't change it in normal circumstances. The reason I added this was as a precaution, just in case that in the future a new hardware is released that supports a smaller number of fragments, although such a situation is unlikely to occur. Furthermore, I noticed that the parameter NumberOfHandledRXPacketsInDPC is also not present in the INF, which led me to believe that it was a design for internal parameters. That's another reason why I proceeded that way previously.

Reply:
@ybendito I have another question. If I don't use this configuration, can I just use this macro in the places where it's needed, or do I still need to do something like this? pConfiguration->MaxFragmentsInOneNB = MAX_FRAGMENTS_IN_ONE_NB

Reply:
@zjmletang I hope you'll get compiler error on "#define a->b c". Just assign it instead of getting from configuration. Later we may invent something useful with it.